src/components/Article.jsx

```diff
@@ -18,6 +20,14 @@ const Article = () => {
         })
     }, [article_id])
 
+    useEffect(() => {
```

Comment:
both of these useEffects share the same loading state which could lead to some bugs, it would be rectified if these were its own component